src/rpc/handlers/LedgerEntry.cpp

```diff
@@ -37,6 +37,12 @@ doLedgerEntry(Context const& context)
         if (!key.parseHex(request.at(JS(index)).as_string().c_str()))
             return Status{Error::rpcINVALID_PARAMS, "malformedIndex"};
     }
+    else if (request.contains(JS(ledger_index)))
```

Comment:
I'm not sure if your code does what was planned or not but i noticed that the whole block is using if-else. this means that if request contains `index` it will never even have a chance to run your code, even if everything was fine with the `index` param. And the same for all other checks down the line. I think this is a bug, wdyt @cjcobb23 ?

Reply:
@godexsoft that's what I suspected as well. That's initially why I had that block of code atop parsing index and ledger_index concurrently to handle the edge case of both fields being present but malformed some other way, whereas the single conditionals could be part of the general case of just that parameter being available